crates/editor/src/signature_help_popover.rs

```diff
@@ -0,0 +1,553 @@
+use crate::{Editor, EditorStyle};
+use gpui::{
+    div, AnyElement, FontWeight, InteractiveElement, IntoElement, MouseButton, ParentElement,
+    Pixels, Size, StatefulInteractiveElement, Styled, ViewContext, WeakView,
+};
+use language::markdown::{MarkdownHighlight, MarkdownHighlightStyle};
+use language::ParsedMarkdown;
+use lsp::SignatureHelp;
+use std::ops::Range;
+use ui::StyledExt;
+use workspace::Workspace;
+
+const SIGNATURE_HELP_HIGHLIGHT: MarkdownHighlight =
+    MarkdownHighlight::Style(MarkdownHighlightStyle {
+        italic: false,
+        underline: true,
+        strikethrough: false,
+        weight: FontWeight::EXTRA_BOLD,
+    });
+
+const SIGNATURE_HELP_OVERLOAD_HIGHLIGHT: MarkdownHighlight =
+    MarkdownHighlight::Style(MarkdownHighlightStyle {
+        italic: true,
+        underline: false,
+        strikethrough: false,
+        weight: FontWeight::NORMAL,
+    });
+
+#[derive(Clone, Debug)]
+pub struct SignatureHelpPopover {
+    pub parsed_content: ParsedMarkdown,
+}
+
+/// create_signature_help_markdown_string generates the markdown text that is displayed in the `SignatureHelp` window.
+pub fn create_signature_help_markdown_string(
+    signature_help: SignatureHelp,
+) -> Option<(String, Vec<(Range<usize>, MarkdownHighlight)>)> {
+    let (signature_information, maybe_active_signature, maybe_active_parameter) = (
+        signature_help.signatures,
+        signature_help.active_signature,
+        signature_help.active_parameter,
+    );
+
+    let function_options_count = signature_information.len();
+
+    let maybe_signature_information = maybe_active_signature
+        .and_then(|active_signature| signature_information.get(active_signature as usize));
+    let signature_information = if let Some(signature_information) = maybe_signature_information {
```

Comment:
NIT: Another block we can fight: `maybe_signature_information.or_else(|| signature_information.first())`

Reply:
thank you!
i fixed it on 4598a4d32